File: anul2/semestrul1/map/lab2/decorator/DelayTaskRunner.java
package decorator;

public class DelayTaskRunner extends AbstractTaskRunner {
    public DelayTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        try {
            System.out.println("Waiting for task to finish...");
            Thread.sleep(3000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        super.executeOneTask();
    }

}

File: Third Semester - Advanced Programming Methods/Lab10/decorator/DelayTaskRunner.cs
using lab10.model;

namespace lab10.decorator;

public class DelayTaskRunner: AbstractTaskRunner
{
    public DelayTaskRunner(ITaskRunner taskRunner): base(taskRunner)
    {
        
    }

    public override void ExecuteOneTask()
    {
        try
        {
            Console.WriteLine("Waiting for task to finish...");
            Thread.Sleep(3000);
        }
        catch (ThreadInterruptedException)
        {
            throw new Exception("The task runner is interrupted.");
        }
        base.ExecuteOneTask();
    }
}